src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -102,8 +102,22 @@ public VSSolutionManager()
                 ErrorHandler.ThrowOnFailure(hr);
             }
 
-            UserAgent.SetUserAgentString(
-                new UserAgentStringBuilder().WithVisualStudioSKU(VSVersionHelper.GetFullVsVersionString()));
+            // Allow this constructor to be invoked from either the UI or a worker thread
+            if (ThreadHelper.CheckAccess())
```

Comment:
This shouldn't be needed. From what I have been told starting a JTF while already on the UI thread is a noop and very fast.